eng/targets/TargetFrameworks.props

```diff
@@ -18,7 +18,7 @@
     <NetVSCode>net9.0</NetVSCode>
     <NetVSShared>net8.0</NetVSShared>
     <NetRoslynBuildHostNetCoreVersion>net6.0</NetRoslynBuildHostNetCoreVersion>
-    <NetRoslynNext>net9.0</NetRoslynNext>
+    <NetRoslynNext>net10.0</NetRoslynNext>
```

Comment:
The only projects using this value are now the `Emit3` unit tests, and `BuildBoss`.